specs/latest/1.0/index.html

```diff
@@ -919,6 +926,50 @@ <h4>Context creation parameters</h4>
                 </p>
               </div>
             </dd>
+        <dt><span class=prop-value>pixelFormat</span></dt>
+            <dd>
+              <p>
+                If defined, require a backbuffer of a specific effective internal format.
+                This is a requirement, and if the requested format is not availible, context creation will fail.
+              </p>
+              <p>
+                The allowed values are:
+              </p>
+              <dl><dd>
+                  <p>
+                    <em>undefined</em>
+                  </p>
+                  <p>
+                    If not defined, choose implicitly based on <code>alpha</code>.
+                    This is the historical behavior.
+                  </p>
+                  <p>
+                    <em>RGBA8</em>
+                  </p>
+                  <p>
+                    Explicitly require an RGBA8 backbuffer.
+                    This is similar to <code>{alpha: true, pixelFormat: "implicit"}</code>.
+                    This is the same effective format as a renderbuffer or texture specified with a <code>sizedFormat</code> or <code>internalFormat</code> of <code>RGBA8</code>,
+                    or a texture specified with <code>{internalFormat: RGBA, unpackFormat: RGBA, unpackType: UNSIGNED_BYTE}</code>.
+                  </p>
+                  <p>
+                    <em>SRGB8_ALPHA8</em>
+                  </p>
+                  <p>
+                    Explicitly require an SRGB8_ALPHA8 "sRGB-encoded" backbuffer.
+                    This is the same effective format as a renderbuffer or texture specified with a <code>sizedFormat</code> or <code>internalFormat</code> of <code>SRGB8_ALPHA8</code>,
+                    or a texture specified with <code>{internalFormat: SRGB_ALPHA, unpackFormat: SRGB_ALPHA, unpackType: UNSIGNED_BYTE}</code>.
+                  </p>
+                  <p>
+                    <em>RGBA16F</em>
+                  </p>
+                  <p>
+                    Explicitly require an RGBA16F half-precision floating point backbuffer.
+                    This is the same effective format as a renderbuffer or texture specified with a <code>sizedFormat</code> or <code>internalFormat</code> of <code>RGBA16F</code>,
+                    or a texture specified with <code>{internalFormat: RGBA, unpackFormat: RGBA, unpackType: HALF_FLOAT_OES}</code>.
```

Comment:
WebGL 1.0 doesn't have floating-point textures by default which makes exposing this pixel format there doubly complicated. If we're moving forward with exposing this on WebGL 1.0 then this should auto-enable OES_texture_float and EXT_color_buffer_half_float (and if they aren't available, this isn't supported). WebGL 2.0 would auto-enable EXT_color_buffer_half_float.

Documentation for `readPixels` needs to be updated in some way - if this is the pixel format, then the `RGBA`/`UNSIGNED_BYTE` isn't supported for readbacks from the default back buffer, `RGBA`/`FLOAT` are.

Reply:
I don't think these should auto-enable. I think auto-enable is *more* tricky than not.
Internally Gecko's WebGL only has extension-checking logic at resource creation. Once it's created (like the backbuffer), even if the extensions aren't enabled, it's smooth sailing to use it however.

Reply:
Chromium will create the resources for the back buffer before the end user has a chance to enable any extensions with the newly-created context. I don't know how we can make this form of the API work without auto-enabling of extensions. Creating the back buffer with these formats will implicitly enable the formats for applications - nearly all of the WebGL validation logic is at a lower level in Chromium's implementation, and we would not want to put back in manual validation of certain internal formats.

Let's discuss on #3222 though since that may sidestep this issue completely.